src/tests/SFA.DAS.EmployerIncentives.Commands.UnitTests/CompleteEarningsCalculation/Handlers/WhenHandlingCompleteEarningsCalculationCommand.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using AutoFixture;
+using FluentAssertions;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.EmployerIncentives.Commands.CompleteEarningsCalculation;
+using SFA.DAS.EmployerIncentives.Commands.Persistence;
+using SFA.DAS.EmployerIncentives.Commands.Types.IncentiveApplications;
+using SFA.DAS.EmployerIncentives.Data;
+using SFA.DAS.EmployerIncentives.Data.Models;
+using SFA.DAS.EmployerIncentives.Domain.IncentiveApplications.Models;
+using SFA.DAS.EmployerIncentives.UnitTests.Shared.AutoFixtureCustomizations;
+using Apprenticeship = SFA.DAS.EmployerIncentives.Domain.IncentiveApplications.Apprenticeship;
+
+namespace SFA.DAS.EmployerIncentives.Commands.UnitTests.CompleteEarningsCalculation.Handlers
+{
+    public class WhenHandlingCompleteEarningsCalculationCommand
+    {
+        private CompleteEarningsCalculationCommandHandler _sut;
+        private Mock<IQueryRepository<IncentiveApplicationApprenticeship>> _mockQueryRepository;
+        private Mock<IIncentiveApplicationDomainRepository> _mockIncentiveApplicationDomainRepository;
+        private Fixture _fixture;
+        private Guid _apprenticeshipIncentiveId;
+
+        [SetUp]
+        public void Arrange()
+        {
+            _fixture = new Fixture();
+            _fixture.Customize(new IncentiveApplicationCustomization());
+            _apprenticeshipIncentiveId = _fixture.Create<Guid>();
+
+            _mockQueryRepository = new Mock<IQueryRepository<IncentiveApplicationApprenticeship>>();
+            
+            _mockIncentiveApplicationDomainRepository = new Mock<IIncentiveApplicationDomainRepository>();
+
+            _sut = new CompleteEarningsCalculationCommandHandler(_mockQueryRepository.Object, _mockIncentiveApplicationDomainRepository.Object);
+        }
+
+        [Test]
+        public async Task Then_the_earnings_calculated_flag_is_updated_for_matching_apprenticeship()
+        {
+            //Arrange
+            var incentiveApplication = _fixture.Create<IncentiveApplication>();
+            var firstApprenticeship = incentiveApplication.Apprenticeships.First();
+            firstApprenticeship.Id = _apprenticeshipIncentiveId;
+            var apprenticeshipModel = _fixture.Build<ApprenticeshipModel>().With(x => x.Id, _apprenticeshipIncentiveId)
+                .Without(x => x.EarningsCalculated).Create();
+            var apprenticeshipModel2 = _fixture.Create<ApprenticeshipModel>();
+
+            var domainIncentiveApplication = _fixture.Create<Domain.IncentiveApplications.IncentiveApplication>();
+
+            domainIncentiveApplication.SetApprenticeships(new List<Apprenticeship> { Apprenticeship.Create(apprenticeshipModel), Apprenticeship.Create(apprenticeshipModel2) });
```

Comment:
I think it would be better to use AuthoFixture to create the ApplicationModel as required and then use the GetExisting factory method to create the domain object.  Then there is no dependency/knowledge about setting up the domain object in the tests.